indico/modules/events/abstracts/util.py

```diff
@@ -227,6 +229,52 @@ def build_default_email_template(event, tpl_type):
     return tpl
 
 
+def generate_spreadsheet_from_abstracts(abstracts, static_item_ids, dynamic_items):
+    """Generates a spreadsheet data from a given abstract list.
+
+    :param abstracts: The list of abstracts to include in the file
+    :param static_item_ids: The abstract properties to be used as columns
+    :param dynamic_items: Contribution fields as extra columns
+    """
+    field_names = ['ID', 'Title']
+    static_item_mapping = OrderedDict([
+        ('state', ('State', lambda x: x.state.title)),
```

Comment:
Performance, as `attrgetter` is implemented in C. Not that it matters much in this particular case.